explorer/interpreter/interpreter.cpp

```diff
@@ -663,6 +671,49 @@ auto Interpreter::InstantiateWitness(Nonnull<const Witness*> witness)
   return cast<Witness>(value);
 }
 
+auto Interpreter::InstantiateClassWithBase(
+    Nonnull<const StructValue*> init_struct,
+    Nonnull<const NominalClassType*> class_type, SourceLocation source_loc)
+    -> ErrorOr<Nonnull<NominalClassValue*>> {
+  std::vector<NamedValue> struct_values;
+  std::optional<Nonnull<const NominalClassValue*>> base_instance;
+  for (const auto& field : init_struct->elements()) {
+    if (field.name == NominalClassValue::BaseField) {
+      CARBON_CHECK(class_type->base().has_value())
+          << "Invalid 'base' field for class '"
+          << class_type->declaration().name() << "' without base class.";
+      switch (field.value->kind()) {
+        case Value::Kind::NominalClassValue: {
+          base_instance = cast<NominalClassValue>(field.value);
+          break;
+        }
+        case Value::Kind::StructValue: {
+          CARBON_ASSIGN_OR_RETURN(
+              auto base,
+              Convert(field.value, class_type->base().value(), source_loc));
+          base_instance = cast<NominalClassValue>(base);
+          break;
+        }
+        default: {
+          CARBON_FATAL()
+              << "Unhandled base value type while initializing class '"
+              << class_type->declaration().name() << "' from a struct";
+        }
+      }
```

Comment:
It would seem simpler to unconditionally call `Convert` here, as you do in the `StructValue` case:
```suggestion
      CARBON_ASSIGN_OR_RETURN(
          auto base,
          Convert(field.value, class_type->base().value(), source_loc));
      base_instance = cast<NominalClassValue>(base);
```
Would that work? If so, that'd make it easier to add new kinds of built-in conversion to class type -- they'd get picked up here automatically.

Reply:
Good call, it does, much nicer :)